### ISSUE TITLE
Deploy TheCombine as Kubernetes cluster

### ISSUE THREAD

**Issue by jmgrady** [sillsdev/TheCombine#1093]
Investigate the cost/benefit of deploying _TheCombine_ on Kubernetes (MicroK8s on the NUCs).

**Comment by johnthagen** [comment]
One angle from a developer perspective is that Docker Desktop (for Windows and macOS) comes with support for starting a single-node cluster out of the box. This could reduce the necessary install steps towards testing on K8s:

https://www.docker.com/products/kubernetes

<img width="932" alt="Screenshot 2021-04-23 114404" src="https://user-images.githubusercontent.com/10340167/115896250-384b3780-a429-11eb-9de5-fb75caa05f28.png">

**Comment by johnthagen** [comment]
Another interesting angle is to keep a Docker Compose config as a source of truth and translate it to K8s on install using Kompose

- https://kubernetes.io/docs/tasks/configure-pod-container/translate-compose-kubernetes/
- https://kompose.io/

**Comment by jmgrady** [comment]
@johnthagen 
Those are good suggestions.
Regarding using Kubernetes in Docker Desktop, there are many options for local, single-node deployment of kubernetes. `kind`, `minikube`, and `microk8s` are ones that I know of.  I have been looking at `minikube` and `microk8s` and am leaning towards `microk8s` for the following reasons:
1. `microk8s` is available for Linux, Mac OS X, and Windows;
2. `microk8s` is usable for production, i.e., we can run it on the NUC; `minikube` is not recommended for production;
Even if we select microk8s for production, it does not mean that developers could not use Kubernetes that comes with Docker Desktop or `minikibe`.  However, my experience with the two I have tried shows that they are not the same - they have different add-ons, different methods for configuring `kubectl`, and so on, so it would be good to standardize.

Regarding `kompose`, I got things started by using `kompose` to generate the files that were needed and it was very helpful.  I don't know if it can be used in a production workflow though because some of the things I need to create the cluster are not available in the `docker-compose.yml` file (& the files it references).  I only briefly looked at it but the parts of the generated files that need tweaking (so far) are:
1. `kompose` sets the volume size for all persistent storage for 100MiB.  This is not sufficient for the database or the backend.
2. The environment variables are put into a configMap but it does not separate secret data (e.g. SMTP login info) from non-secret data.
3. I do not know of a way to set up "image pull secrets" in the `docker-compose` file so that the images can be pulled from the AWS Elastic Container Registry (or other private repo).  We currently need to have the person installing the software do the login before they can do a `docker-compose pull`.

That's what I've learned so far.  I think we should discuss what types of problems we need to solve in a Kubernetes environment.  The main change to the software is dividing things into containers.  I impression is that `docker-compose` would provide the containers but would be easier for tracking down any problems.  Any problems that only appear in the Kubernetes environment are likely to be because we are scaling up the application or because it is distributed across multiple nodes.
